Tasks/XamarinAndroid/xamarinandroid.ts

```diff
@@ -47,7 +47,11 @@ else {
 //find xbuild location to use
 var xbuildToolPath = tl.which('xbuild');
 if(xbuildLocation) {
-    xbuildToolPath = xbuildLocation + '/xbuild';
+    xbuildToolPath = path.join(xbuildLocation, 'xbuild');
+    if (!tl.exist(xbuildToolPath)) {
+        xbuildToolPath = path.join(xbuildLocation, 'xbuild.exe');
```

Comment:
Unfortunately, there isn't a way to mock that OS check. That's why I tried to avoid that here. The TaskLib really needs a method that allows the task to check the OS (tl.isWin). That way it could be mocked and all paths can be fully tested.